libcudacxx/.upstream-tests/test/support/overrun_guard.h

```diff
@@ -0,0 +1,75 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef TEST_OVERRUN_GUARD
+#define TEST_OVERRUN_GUARD
+
+#include <cuda/std/type_traits>
+
+template<typename T>
```

Comment:
At some point during this work I managed to do a dumb and write over some bytes beyond the variable, which results in a silly behavior of an endless hang if you happen to overwrite the barrier; fun times. Hopefully this will catch the more likely cases of off-by-ones (that's what I did, didn't subtract 1 from the first set bit index when turning that index into an actual alignment value) with a reasonable assert message instead of a hang in `arrive_and_wait`.

Reply:
Oh I mean in the file for future reference ;)